lib/Runtime/Library/JavascriptLibrary.cpp

```diff
@@ -6153,7 +6153,16 @@ namespace Js
     {
         Assert(scriptContext->GetConfig()->IsES6GeneratorsEnabled());
         DynamicType* generatorType = CreateGeneratorType(prototype);
-        return RecyclerNew(this->GetRecycler(), JavascriptGenerator, generatorType, args, scriptFunction);
+#if GLOBAL_ENABLE_WRITE_BARRIER
+        if (CONFIG_FLAG(ForceSoftwareWriteBarrier))
+        {
+            return RecyclerNewFinalized(this->GetRecycler(), JavascriptGenerator, generatorType, args, scriptFunction);
+        }
+        else
+#endif
+        {
+            return RecyclerNew(this->GetRecycler(), JavascriptGenerator, generatorType, args, scriptFunction);
+        }
```

Comment:
Consider moving this into a function JavascriptGenerator::New (and make constructor private) so that the logic at the same place.